app/services/course/assessment/question/programming_import_service.rb

```diff
@@ -0,0 +1,105 @@
+# Imports the provided programming package into the question. This evaluates the package to
+# obtain the set of tests, as well as extracts the templates from the package to be stored
+# together with the question.
+class Course::Assessment::Question::ProgrammingImportService
+  class << self
+    # Imports the programming package into the question.
+    #
+    # @param [Course::Assessment::Question::Programming] question The programming question for
+    #   import.
+    # @param [Course::Assessment::ProgrammingPackage] package The package containing the
+    #   tests and template files.
+    #
+    # @overload import(question, package)
+    #   @param [Course::Assessment::Question::Programming] question The programming question for
+    #     import.
+    #   @param [String] package The path to the package containing the tests and template files.
+    def import(question, package)
+      package = Course::Assessment::ProgrammingPackage.new(package) unless \
+        package.is_a?(Course::Assessment::ProgrammingPackage)
+      new(question, package).send(:import)
+    end
+  end
+
+  private
+
+  # Creates a new service import object.
+  #
+  # @param [Course::Assessment::Question::Programming] question The programming question for import.
+  # @param [Course::Assessment::ProgrammingPackage] package The package containing the tests and
+  #   template files.
+  def initialize(question, package)
+    @question = question
+    @package = package
+  end
+
+  # Imports the templates and tests found in the package.
+  def import
+    template_files = nil
+    template_import_thread = Thread.new { template_files = import_template_files }
+    evaluation_result = evaluate_package
```

Comment:
And why need to run the evaluation ? there is no student code right in the package right ? only the template files.